consensus/p2p/statemachine/transition.go

```diff
@@ -49,14 +45,19 @@ type Transition interface {
 	) (*FinState, error)
 }
 
-type transition struct{}
+type transition[V types.Hashable[H], H types.Hash, A types.Addr] struct {
+	validator validator.Validator[V, H, A]
```

Comment:
This is important:
Transition shouldn't have the state, but only the necessary dependencies to calculate the transitions. States should be in the states like `InitialState`,....
Why:
`Transition` is supposed to be shared between multiple proposal stream. Putting the states here mean it's a mutable state, which means there must be a new instance created for every proposal stream.